recipes/conversation_view/attachment_carousel/attachment_carousel.vue

```diff
@@ -0,0 +1,263 @@
+    <dt-button
+      v-if="showLeftArrow && showCarouselArrows"
+      tabindex="-1"
+      class="dt-attachment-carousel--left-arrow d-ba d-baw1 d-bc-default"
+      circle
+      size="xs"
+      importance="clear"
+      :aria-label="leftArrowAriaLabel"
+      @click="leftScroll"
+    >
+      <template #icon>
+        <dt-icon
+          name="arrow-left"
+          size="100"
+        />
+      </template>
+    </dt-button>
+    <dt-button
+      v-if="showRightArrow && showCarouselArrows"
+      tabindex="-1"
+      class="dt-attachment-carousel--right-arrow d-ba d-baw1 d-bc-default"
+      circle
+      size="xs"
+      importance="clear"
+      :aria-label="rightArrowAriaLabel"
+      @click="rightScroll"
+    >
+      <template #icon>
+        <dt-icon
+          name="arrow-right"
+          size="100"
+        />
+      </template>
+    </dt-button>
+  </div>
+</template>
+
+<script>
+import { DtIcon } from '@/components/icon';
+import { DtImageViewer } from '@/components/image_viewer';
+import { DtButton } from '@/components/button';
+import {} from './attachment_carousel_constants';
+
+export default {
+  name: 'DtRecipeAttachmentCarousel',
+
+  components: {
+    DtButton,
+    DtImageViewer,
+    DtIcon,
+  },
+
+  mixins: [],
+
+  /* inheritAttrs: false is generally an option we want to set on library
+    components. This allows any attributes passed in that are not recognized
+    as props to be passed down to another element or component using v-bind:$attrs
+    more info: https://vuejs.org/v2/api/#inheritAttrs */
+  // inheritAttrs: false,
+
+  props: {
+    /**
+     * media - object array of media objects
+     * @type {Array}
+     *
+     * Object: {
+     *  path: String,
+     *  altText: String | null,
+     * }
+     */
+    mediaList: {
+      type: Array,
+      default: () => [],
+    },
+
+    /**
+     * Descriptive label for the attachment element
+     */
+    attachmentAriaLabel: {
+      type: String,
+      required: true,
+      default: '',
+    },
+
+    closeAriaLabel: {
+      type: String,
+      required: true,
+      default: '',
+    },
+
+    clickToOpenAriaLabel: {
+      type: String,
+      required: true,
+      default: '',
+    },
+
+    leftArrowAriaLabel: {
+      type: String,
+      required: true,
+      default: '',
+    },
+
+    rightArrowAriaLabel: {
+      type: String,
+      required: true,
+      default: '',
+    },
+  },
+
+  emits: [
+    /**
+     * Emitted when popover is shown or hidden
+     *
+     * @event remove-image
+     * @type {Number}
+     */
+    'remove-image',
+  ],
+
+  data () {
+    return {
+      showCloseButton: {},
+      showCarouselArrows: false,
+      showRightArrow: true,
+      showLeftArrow: false,
+      isMounted: false,
+    };
+  },
+
+  computed: {
+
+  },
+
+  mounted: function () {
+    this.showLeftArrow = this.$refs.carousel.scrollLeft > 0;
+    this.showRightArrow = this.$refs.carousel.scrollWidth > this.$refs.carousel.clientWidth;
```

Comment:
can you just call handleScroll?

Reply:
the showRightArrow is different in handleScroll. Once mounted, we check if there is more items than what can fit in the width. While on scroll we check if we reached the last item.